test/hotspot/jtreg/compiler/loopopts/parallel_iv/TestParallelIvInIntCountedLoop.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.loopopts.parallel_iv;
+
+import compiler.lib.ir_framework.Argument;
+import compiler.lib.ir_framework.Arguments;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+
+import java.util.Random;
+import java.util.function.Function;
+
+/**
+ * @test
+ * @bug 8328528
+ * @summary test the long typed parallel iv replacing transformation for int counted loop
+ * @library /test/lib /
+ * @run driver compiler.loopopts.parallel_iv.TestParallelIvInIntCountedLoop
+ */
+public class TestParallelIvInIntCountedLoop {
+    private static final int stride;
+    private static final int stride2;
+
+    static {
+        // stride2 must be a multiple of stride and must not overflow for the optimization to work
+        stride = new Random().nextInt(1, Integer.MAX_VALUE / 16);
+        stride2 = stride * new Random().nextInt(1, 16);
+    }
+
+    public static void main(String[] args) {
+        TestFramework.run();
+        testCorrectness();
+    }
+
+    /*
+     * The IR framework can only test against static code, and the transformation relies on strides being constants to
+     * perform constant propagation. Therefore, we have no choice but repeating the same test case multiple times with
+     * different numbers.
+     *
+     * For good measures, a randomly initialized static final stride and stride2 is also tested.
+     */
+    @Test
+    @Arguments(values = {Argument.RANDOM_ONCE})
+    @IR(failOn = {IRNode.LOOP, IRNode.COUNTED_LOOP})
+    private static int testIntCountedLoopWithIntIV(int stop) {
+        int a = 0;
+        for (int i = 0; i < stop; i++) {
+            a += 1;
+        }
+
+        return a;
+    }
```

Comment:
You have `failOn` for every test. I'm worried that something with the test could be so wrong that we just don't have a `LoopNode` for some completely wrong reason.
Can you please add a "control test", that actually has a simple loop, and where you can find it with an `@IR` check?